### PR TITLE
CTAS with paranthesized expression

### PR DESCRIPTION
**Pull request by barakalon**
Addresses https://github.com/tobymao/sqlglot/issues/117

@tobymao - take a critical eye at this. The WITH/SELECT recursion is getting kind of messy, and I don't have a good mental catalog of what's valid in various dialects.

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -480,7 +480,14 @@ def _parse_create(self):
                 this if isinstance(this, exp.Schema) else None
             )
             if self._match(TokenType.ALIAS):
+                parenthesized = self._match(TokenType.L_PAREN)
```
Comment (tobymao) [inline sqlglot/parser.py:483]: i don't think this is correct, it doesn't handle recursive parens

Comment (tobymao) [reply sqlglot/parser.py:483]: i have a fix

**Comment by tobymao** [comment]
@barakalon https://github.com/tobymao/sqlglot/commit/9e4b66cdcb553e1ecaca3f20a269298d8b1c7b33

the issue was that on paren, we look for a select instead of a with, with is a superset of select, so it's safe

**Comment by barakalon** [comment]
Sweet. I had tried that out, but the extra recursion felt scary to me. Glad you took care of it :)

**Pull request closed without merging**